src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,571 @@
+
+        if (vat.live() == 0) {
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(ilks[ilk].culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED,
+                currentAssets
+            );
+        } else if (ilks[ilk].tic != 0 || !pool.active() || !ilks[ilk].plan.active()) {
+            // pool caged
+            _unwind(
+                ilk,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                ilks[ilk].culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL,
+                currentAssets
+            );
+        } else {
+            // Determine if it needs to unwind due to debt ceilings
+            (uint256 Art,,, uint256 line,) = vat.ilks(ilk);
+            uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 toUnwind;
+            if (Art > lineWad) {
+                toUnwind = Art - lineWad;
+            }
+            if (debt > Line) {
+                toUnwind = _max(toUnwind, _divup(debt - Line, RAY));
+            }
+
+            // Determine if it needs to unwind due plan
+            uint256 targetAssets = ilks[ilk].plan.getTargetAssets(currentAssets);
+            if (targetAssets < currentAssets) {
+                toUnwind = _max(toUnwind, currentAssets - targetAssets);
+            }
+
+            if (toUnwind > 0) {
+                _unwind(
+                    ilk,
+                    pool,
+                    toUnwind,
+                    availableAssets,
+                    Mode.NORMAL,
+                    currentAssets
+                );
+            } else {
+                // All the subtractions are safe as otherwise toUnwind is > 0
+                uint256 toWind = targetAssets - currentAssets;
+                toWind = _min(toWind, lineWad - Art);
+                toWind = _min(toWind, (Line - debt) / RAY);
+                // Determine if the pool limits our total deposits
+                toWind = _min(toWind, pool.maxDeposit());
+                _wind(ilk, pool, toWind);
+            }
+        }
+    }
+
+    /**
+        @notice Collect interest and send to vow.
+        Total collected will be constrained by the number of assets available to
+        be withdrawn from the pool.
+        @param ilk bytes32 of the D3M ilk name
+    */
+    function reap(bytes32 ilk) external {
+        D3MPoolLike pool = ilks[ilk].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(ilks[ilk].tic == 0, "DssDirectDepositHub/pool-not-live");
```

Comment:
Making a note here for myself to come back to once I understand other things better: should this `require` also check `pool.active()` and `plan.active()`, as is done in `exec`? Alternatively, is this `require` actually necessary? Is it possibly more appropriate for it to check `culled`?

Reply:
Hmmmm tentatively thinking 1) yes the `require` is appropriate, and 2) should include `active` checks for consistency (unclear to me what "inactive" is supposed to signify for a pool/plan though).